packages/connection-storage/src/utils.ts

```diff
@@ -38,3 +39,20 @@ export function deleteCompassAppNameParam(
     },
   };
 }
+
+export const throwIfAborted = (signal?: AbortSignal) => {
+  if (signal?.aborted) {
+    const err = signal.reason ?? new Error('This operation was aborted.');
+    throw err;
+  }
+};
```

Comment:
This should probably be put into `packages/compass-utils/src/cancellable-promise.ts`, and use the existing `createCancelError()` method